bot/exts/evergreen/profile_pic_modification/_effects.py

```diff
@@ -0,0 +1,134 @@
+        bufferedio.seek(0)
+
+        return discord.File(bufferedio, filename="modified_avatar.png")
+
+    @staticmethod
+    def closest(x: t.Tuple[int, int, int]) -> t.Tuple[int, int, int]:
+        """
+        Finds the closest easter colour to a given pixel.
+
+        Returns a merge between the original colour and the closest colour.
+        """
+        r1, g1, b1 = x
+
+        def distance(point: t.Tuple[int, int, int]) -> t.Tuple[int, int, int]:
+            """Finds the difference between a pastel colour and the original pixel colour."""
+            r2, g2, b2 = point
+            return (r1 - r2) ** 2 + (g1 - g2) ** 2 + (b1 - b2) ** 2
+
+        closest_colours = sorted(EASTER_COLOURS, key=lambda point: distance(point))
+        r2, g2, b2 = closest_colours[0]
+        r = (r1 + r2) // 2
+        g = (g1 + g2) // 2
+        b = (b1 + b2) // 2
+
+        return r, g, b
+
+    @staticmethod
+    def crop_avatar_circle(avatar: Image) -> Image:
+        """This crops the avatar given into a circle."""
+        mask = Image.new("L", avatar.size, 0)
+        draw = ImageDraw.Draw(mask)
+        draw.ellipse((0, 0) + avatar.size, fill=255)
+        avatar.putalpha(mask)
+        return avatar
+
+    @staticmethod
+    def crop_ring(ring: Image, px: int) -> Image:
+        """This crops the given ring into a circle."""
+        mask = Image.new("L", ring.size, 0)
+        draw = ImageDraw.Draw(mask)
+        draw.ellipse((0, 0) + ring.size, fill=255)
+        draw.ellipse((px, px, 1024-px, 1024-px), fill=0)
+        ring.putalpha(mask)
+        return ring
+
+    @staticmethod
+    def pridify_effect(
+        image: Image,
+        pixels: int,
+        flag: str
+    ) -> Image:
+        """Applies the pride effect to the given image."""
+        image = image.resize((1024, 1024))
+        image = PfpEffects.crop_avatar_circle(image)
+
+        ring = Image.open(Path(f"bot/resources/pride/flags/{flag}.png")).resize((1024, 1024))
+        ring = ring.convert("RGBA")
+        ring = PfpEffects.crop_ring(ring, pixels)
+
+        image.alpha_composite(ring, (0, 0))
+        return image
+
+    @staticmethod
+    def eight_bitify_effect(image: Image) -> Image:
+        """Applies the 8bit effect to the given image."""
+        image = image.resize((32, 32), resample=Image.NEAREST).resize((1024, 1024), resample=Image.NEAREST)
+        return image.quantize()
+
+    @staticmethod
+    def easterify_effect(image: Image, overlay_image: Image = None) -> Image:
+        """Applies the easter effect to the given image."""
+        if overlay_image:
+            ratio = 64 / overlay_image.height
+            overlay_image = overlay_image.resize((
+                round(overlay_image.width * ratio),
+                round(overlay_image.height * ratio)
+            ))
+            overlay_image = overlay_image.convert("RGBA")
+        else:
+            overlay_image = overlay_image = Image.open(Path("bot/resources/easter/chocolate_bunny.png"))
```

Comment:
I didn’t understand this line. Why are you double assigning the same variables to the same value?